src/main/java/com/openlattice/datastore/services/EdmService.java

```diff
@@ -684,19 +684,16 @@ public PropertyType getPropertyType( FullQualifiedName propertyType ) {
     @Override
     public Map<UUID, PropertyType> getPropertyTypesForEntitySet( UUID entitySetId ) {
 
-        Optional<UUID> maybeEtId = (Optional<UUID>) entitySets.executeOnKey( entitySetId, new GetEntityTypeFromEntitySetEntryProcessor() );
-        if ( maybeEtId.isEmpty() ) {
+        UUID maybeEtId = (UUID) entitySets.executeOnKey( entitySetId, new GetEntityTypeFromEntitySetEntryProcessor());
```

Comment:
I think the cast should be after the null check

Reply:
Doesn't it not matter since Set<UUID> and UUID are both able to safely be null?

Reply:
Ah, yeah, you're right! I thought it will nullpointer